File: LeetCode75/TwoPointers/283_move_zeroes/moveZeroes.py
from typing import List


class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        left = 0

        for right in range(len(nums)):
            if nums[left] == 0:
                nums[left], nums[right] = nums[right], nums[left]
            if nums[left] != 0:
                left += 1

File: 283_move_zeroes/move_zeroes.rs
impl Solution {
    pub fn move_zeroes(nums: &mut Vec<i32>) {
        let mut l = 0;

        for r in 0..nums.len() {
            if nums[r] != 0 {
                nums.swap(l, r);
                l += 1;
            }
        }
    }
}